tests/unit/amo/pages/TestAddonReviewList.js

```diff
@@ -1,93 +1,94 @@
-import { shallow } from 'enzyme';
-import * as React from 'react';
+import { waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 import {
+  fetchReview,
   fetchReviews,
   fetchReviewPermissions,
   setAddonReviews,
+  setReview,
   setReviewPermissions,
 } from 'amo/actions/reviews';
 import { setViewContext } from 'amo/actions/viewContext';
-import AddonReviewList, {
-  AddonReviewListBase,
-  extractId,
-  SHOW_ALL_REVIEWS,
-} from 'amo/pages/AddonReviewList';
-import AddonReviewCard from 'amo/components/AddonReviewCard';
-import AddonSummaryCard from 'amo/components/AddonSummaryCard';
-import FeaturedAddonReview from 'amo/components/FeaturedAddonReview';
-import Page from 'amo/components/Page';
-import { getAddonURL } from 'amo/utils';
-import { ErrorHandler } from 'amo/errorHandler';
-import Link from 'amo/components/Link';
-import { reviewListURL } from 'amo/reducers/reviews';
+import { extractId, SHOW_ALL_REVIEWS } from 'amo/pages/AddonReviewList';
 import { DEFAULT_API_PAGE_SIZE, createApiError } from 'amo/api';
-import Paginate from 'amo/components/Paginate';
 import {
-  ADDON_TYPE_EXTENSION,
   ADDON_TYPE_STATIC_THEME,
   CLIENT_APP_FIREFOX,
   SET_VIEW_CONTEXT,
 } from 'amo/constants';
 import { fetchAddon, loadAddon } from 'amo/reducers/addons';
-import Card from 'amo/components/Card';
-import ErrorList from 'amo/components/ErrorList';
-import LoadingText from 'amo/components/LoadingText';
 import {
-  createCapturedErrorHandler,
-  createFakeEvent,
-  createFakeHistory,
-  createFakeLocation,
-  createInternalAddonWithLang,
+  createFailedErrorHandler,
+  createFakeErrorHandler,
+  createHistory,
   createLocalizedString,
-  createStubErrorHandler,
   dispatchClientMetadata,
-  dispatchSignInActions,
+  dispatchSignInActionsWithStore,
   fakeAddon,
-  fakeI18n,
   fakeReview,
-  shallowUntilTarget,
+  getElement,
+  getElements,
+  onLocationChanged,
+  renderPage as defaultRender,
+  screen,
+  within,
 } from 'tests/unit/helpers';
 
 describe(__filename, () => {
   const clientApp = CLIENT_APP_FIREFOX;
+  const defaultSlug = 'reviewed-add-on';
+  const defaultUserId = 123;
   const lang = 'en-US';
   let store;
+  let addon;
+
+  const getLocation = ({ page, reviewId, score, slug = defaultSlug } = {}) => {
+    let queryString = '?';
+    if (page) {
+      queryString = `${queryString}&page=${page}`;
+    }
+    if (score) {
+      queryString = `${queryString}&score=${score}`;
+    }
+    return `/${lang}/${clientApp}/addon/${slug}/reviews/${
+      reviewId ? `${reviewId}/` : ''
+    }${queryString}`;
+  };
```

Comment:
nit, and possibly not even valid but is there not a nicer way to write this?  In python you'd build the parts and `querystring = '&'.join(parts)` or similar.

Reply:
I could do this:

```
    const queryStringArgs = [];
    if (page) {
      queryStringArgs.push(`page=${page}`);
    }
    if (score) {
      queryStringArgs.push(`score=${score}`);
    }
    return `/${lang}/${clientApp}/addon/${slug}/reviews/${
      reviewId ? `${reviewId}/` : ''
    }?${queryStringArgs.join('&')}`;
```

but I'm not sure that's any better. One thing to note is that `reviewId` doesn't become part of the queryString, but rather part of the path.